rbd/rbd_pacific_test.go

```diff
@@ -0,0 +1,49 @@
+// +build !octopus,!nautilus
+
+package rbd
+
+import (
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncryptionFormat(t *testing.T) {
+	conn := radosConnect(t)
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	assert.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+
+	name := GetUUID()
+	testImageSize := uint64(1 << 23) // format requires more than 4194304 bytes
+	options := NewRbdImageOptions()
+	assert.NoError(t,
+		options.SetUint64(ImageOptionOrder, uint64(testImageOrder)))
+	err = CreateImage(ioctx, name, testImageSize, options)
+	assert.NoError(t, err)
+
+	workingImage, err := OpenImage(ioctx, name, NoSnapshot)
+	assert.NoError(t, err)
+
+	var opts EncryptionFormatOptsLUKS1
+	opts.Alg = RBD_ENCRYPTION_ALGORITHM_AES256
+	opts.Passphrase = "test-password"
+	opts.PassphraseSize = uint(len(opts.Passphrase))
+	err = workingImage.EncryptionFormat(RBD_ENCRYPTION_FORMAT_LUKS1, unsafe.Pointer(&opts))
+	assert.NoError(t, err)
+
```

Comment:
can you add some write and read operations too? Does reading without decrypting  fail, or is the read data encrypted?

Reply:
it looks like this function may not be sufficient for reading/writing. Because there's no doc comment/desciprtion in the ceph header file. I searched and found the following from qemu: https://lore.kernel.org/qemu-devel/YJFeV5YwdBRywKAX@redhat.com/t/

There's a description of rbd_encryption_format as well as rbd_encryption_load. The latter appears needed to do IO.

Reply:
The rbd_encryption_load is required for IO assuming you are using librbd.  For users who then mount an encrypted image with librbd/qemu this is not a requirement.